vouch/controllers/metrics_controller.py

```diff
@@ -36,6 +45,50 @@ def __init__(self):
         cert = query_vault(self._vault)
         self.cert_expiration_time = cert.not_valid_after
 
+        # For querying host signing token expiry
+        self.vault_token = CONF['vault_token']
+        self.host_signing_token_expiry = None
+        self.host_signing_token_last_update_time = time.time()
+        self.customer_uuid = os.environ['CUSTOMER_ID']
+        config_url = os.environ.get('CONSUL_HTTP_ADDR', None)
+        token = os.environ.get('CONSUL_HTTP_TOKEN', None)
+        self.consul = Consul(config_url, token=token)
+    
+    def get_host_signing_token_expiry(self):
+        current_time = time.time()
+    
+        try:
+            vault_url = self.consul.kv_get(f'customers/{self.customer_uuid}/vouch/vault/url')
+            host_signing_token = self.consul.kv_get(f'customers/{self.customer_uuid}/vouch/vault/host_signing_token')
+
+            headers = {'X-Vault-Token': host_signing_token}
+
+            LOG.info("Sending request to Vault token lookup-self API...")
+            response = requests.get(f'{vault_url}/v1/auth/token/lookup-self', headers=headers)
+
+            if response.status_code != 200:
```

Comment:
What status code does this Vault API return when the token is expired? probably (403), but maybe confirm once.
So in that case should we return `0` expiry time instead of `None`?
What do we see in prom when the metric value is set to `None`?
Would be good to try and test this expired token case